assets/home/sections/quick-links.js

```diff
@@ -2,21 +2,135 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { Icon, external } from '@wordpress/icons';
+import { useState, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Section from '../section';
+import { Grid, Col } from '../grid';
+
+const quickLinksSpecialMapping = {
+	'sensei://install-demo-course': ( { setTitle, originalTitle, timer } ) => {
+		setTitle( 'Installing' );
+		const installedMessage = 'Installed!';
+		const clear = () => {
+			if ( timer.current ) {
+				clearTimeout( timer.current );
+			}
+			timer.current = 0;
+		};
+		const run = () => {
+			setTitle( ( oldTitle ) => {
+				if ( installedMessage === oldTitle ) {
+					clear();
+					return originalTitle;
+				}
+				if ( ! oldTitle.includes( '...' ) ) {
+					timer.current = setTimeout( run, 500 );
+					return oldTitle + '.';
+				}
+				timer.current = setTimeout( run, 2000 );
+				return installedMessage;
+			} );
+		};
+		clear();
+		timer.current = setTimeout( run, 500 );
+	},
+};
 
 /**
- * Quick Links section component.
+ * Component that shows a Quick Link.
+ *
+ * @param {Object} props      Component props.
+ * @param {Object} props.item The item to show.
+ */
+const QuickLink = ( { item } ) => {
+	let { url, title: originalTitle } = item;
+	const [ title, setTitle ] = useState( originalTitle );
+	const timer = useRef( 0 );
+
+	let onClick = null;
+
+	const callback = quickLinksSpecialMapping[ url ];
```

Comment:
I wonder if it could be a little simpler and more flexible using a separate component for special behaviors. But we could also think a little more about that when integrating this too.

Reply:
Yeah, I'm not sure. To be honest, I do not see right now much more special mappings than the one for importing the demo course, which means that maybe not even the mapping is needed. I only did this to allow to have a cleaner code and separate better the responsibilities, but I'm not sure if we should work much more to split this into another component.. :thinking:

Reply:
I think that's fine! 😉